packages/core/__tests__/client.ts

```diff
@@ -0,0 +1,136 @@
+/// <reference types="jest" />
+import { IMockAdapterOptions, MockAdapter } from '../__mocks__/MockAdapter';
+import * as Sentry from '../index';
+
+const dsn = 'https://username:password@domain/path';
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+describe('Sentry.Client', () => {
+  test('get public/private DSN', () => {
+    const sentry = new Sentry.Client('https://username:password@domain/path');
+    expect(sentry.dsn.getDsn(false)).toBe('https://username@domain/path');
+    expect(sentry.dsn.getDsn(true)).toBe('https://username:password@domain/path');
+    const sentry2 = new Sentry.Client('https://username:password@domain:8888/path');
+    expect(sentry2.dsn.getDsn(false)).toBe('https://username@domain:8888/path');
+    expect(sentry2.dsn.getDsn(true)).toBe('https://username:password@domain:8888/path');
+  });
+  /* tslint:disable */
+  test('invalid DSN', () => {
+    expect(() => {
+      new Sentry.Client('abc');
+    }).toThrow();
+    expect(() => {
+      new Sentry.Client('https://username:password@domain');
+    }).toThrow();
+    expect(() => {
+      new Sentry.Client('//username:password@domain');
+    }).toThrow();
+    expect(() => {
+      new Sentry.Client('123');
+    }).toThrow();
+    try {
+      new Sentry.Client('123');
+    } catch (e) {
+      expect(e instanceof Sentry.SentryError).toBeTruthy();
+    }
+  });
+  /* tslint:enable */
+
+  test('throw error if multiple Adapters', () => {
+    const sentry = Sentry.create(dsn);
+    expect(Sentry.getSharedClient()).toBe(sentry);
+    sentry.use(MockAdapter);
+    expect(() => sentry.use(MockAdapter)).toThrow();
+  });
+
+  test('call install on Adapter', () => {
+    const sentry = new Sentry.Client(dsn);
+    sentry.use(MockAdapter, { testOption: true });
+    const spy1 = jest.spyOn(sentry, 'install');
+    const spy2 = jest.spyOn(sentry.getAdapter(), 'install');
+    sentry.install();
+    expect(spy1).toHaveBeenCalledTimes(1);
+    expect(spy2).toHaveBeenCalledTimes(1);
```

Comment:
In all these tests you rarely test if the arguments are passed in correctly (i.e. the options, I guess?). It would be quite easy to do this using snapshots.